### PR TITLE
MIRK use Interpolation

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

**Comment by avik-pal** [comment]
Run `JuliaFormatter` and rebase. This should be mostly ready to go in after that.

**Comment by ChrisRackauckas** [comment]
@avik-pal were you going to do that?

**Comment by avik-pal** [comment]
@difeizhang

**Comment by difeizhang** [comment]
Sorry. Will do it by Friday.

File: Project.toml [hunk 1]
```diff
@@ -8,16 +8,21 @@ Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
...
+OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
 PreallocationTools = "d236fae5-4411-538c-8e31-a6e3d9e00b46"
 RecursiveArrayTools = "731186ca-8d62-57ce-b412-fbd966d074cd"
 Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
 SciMLBase = "0bca4576-84f4-4d90-8ffe-ffa030f20462"
 Setfield = "efcf1570-3423-57d1-acb7-fd33fddbac46"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 SparseDiffTools = "47a9eef4-7e08-11e9-0b38-333d64bd3804"
+StaticArrays = "90137ffa-7385-5640-81b9-e52037218182"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
```
Comment (avik-pal) [inline Project.toml:25]: These are not meant to be here

File: src/solve/mirk.jl [hunk 2]
```diff
@@ -189,7 +189,12 @@ function construct_nlproblem(cache::MIRKCache{iip}, y::AbstractVector) where {ii
     loss_bc = if iip
         function loss_bc_internal!(resid::AbstractVector, u::AbstractVector, p = cache.p)
             y_ = recursive_unflatten!(cache.y, u)
-            eval_bc_residual!(resid, cache.problem_type, cache.bc, y_, p, cache.mesh)
+            bc_sol_ = DiffEqBase.build_solution(cache.prob,
+                cache.alg,
+                cache.mesh,
+                y_;
+                interp = MIRKInterpolation(cache.mesh, y_, cache)) # build solution @ any time
+            eval_bc_residual!(resid, cache.problem_type, cache.bc, bc_sol_, p, cache.mesh)
```
Comment (avik-pal) [inline src/solve/mirk.jl:197]: Use a conditional and construct this only if it is a multi-point problem

File: Project.toml (hunk 1 above)
Comment (difeizhang) [reply Project.toml]: have restored the Project.toml

File: Project.toml [hunk 3]
```diff
@@ -18,6 +18,7 @@ SciMLBase = "0bca4576-84f4-4d90-8ffe-ffa030f20462"
 Setfield = "efcf1570-3423-57d1-acb7-fd33fddbac46"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
 SparseDiffTools = "47a9eef4-7e08-11e9-0b38-333d64bd3804"
+Static = "aedffcd0-7271-4cad-89d0-dc628f76c6d3"
```
Comment (ChrisRackauckas) [inline Project.toml:21]: Why is this here? We shouldn't rely on Static please.

Comment (difeizhang) [reply Project.toml]: My git is a bit messy. I create a new one and will make a new PR. Shouldn't have static this time.
